features/git-rename-branch/non_feature_branch/with_remote_origin/with_tracking_branch/with_unpushed_changes.feature

```diff
@@ -0,0 +1,34 @@
+Feature: git rename-branch: errors if renaming a non-feature branch that has unpushed changes
+
+  As a developer renaming a non-feature branch that has unpushed changes
+  I should get an error that the given branch is not in sync with its tracking branch
+  So that I don't lose work by accidentally overwriting existing branches.
```

Comment:
Technically we wouldn't lose changes if the branch to rename contains unpushed changes. I'd be open to just go ahead with the rename in this case.